redbiom/commands/summarize.py

```diff
@@ -74,17 +80,70 @@ def summarize_metadata(descending):
         click.echo("%s\t%s" % (idx, val))
 
 
+def _summarize_id(context, category, id):
+    """Summarize the ID over the category"""
+    from redbiom.summarize import category_from_observations
+    res = category_from_observations(context, category, [id], False)
+    res = res.value_counts()
+    counts = {i: c for i, c in zip(res.index, res)}
+    counts['feature'] = id
+    return counts
+
+
+@summarize.command(name='table')
+@click.option('--category', type=str, required=True)
+@click.option('--context', required=True, type=str)
+@click.option('--output', required=False, type=click.Path(exists=False),
+              default=None)
+@click.option('--threads', type=int, default=1)
+@click.option('--verbosity', type=int, default=0)
+@click.option('--table', type=click.Path(exists=True), required=True)
+def summarize_table(category, context, output, threads, verbosity, table):
+    """Summarize all observations in a BIOM table.
+
+    This command will assess, per observation, the number of samples that
+    observation is found in relative to the metadata category specified.
+    """
+    if threads == 1:
+        # TODO: requests.Session is apparently not threadsafe. Basically,
+        # if we just use this as is, with threads, it vomits horribly.
+        # But if you block the ability to spawn a session, it works.
+        import redbiom.util
+        if not redbiom.util.category_exists(category):
+            import sys
+            click.echo("%s is not found" % category, err=True)
+            sys.exit(1)
+
+    import biom
+    table = biom.load_table(table)
+
+    import joblib
+    with joblib.parallel.Parallel(n_jobs=threads, verbose=verbosity) as par:
+        mappings = par(joblib.delayed(_summarize_id)(context, category, id)
+                       for id in table.ids(axis='observation'))
+
+    import pandas as pd
+    df = pd.DataFrame(mappings)
+    df.set_index('feature', inplace=True)
+    df[df.isnull()] = 0
```

Comment:
These are integers, not floats, and zero occurrences are appropriate for similar reasons why zeros are appropriate in an OTU table